### PR TITLE
get units from results

### PR DESCRIPTION
**Pull request by paul-krohn**
cloudwatch-to-graphite wants you to configure the units for each metric. This is un-necessary, and requires that you write a lot of config if you want several metrics that are in different units. This change allows to to not specify the units, so you get all available units (usually 1 per metric) and send all to graphite.

### CODE REVIEW

File: leadbutt.py [hunk 1]
```diff
@@ -133,18 +135,34 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
...
-        )
-        # sys.stderr.write('{} {}\n'.format(options['Count'], len(results)))
-        output_results(results, metric, options)
+        # if the unit is in the yaml config, send it and get only those.
+        # else don't send it, get all the available units, and in output_results,
+        # we'll loop over all the results and send all to graphite.
+        if 'Unit' in metric.keys():
+            unit = metric['Unit']
+        else:
+            unit = None
```
Comment (crccheck) [inline leadbutt.py]: Can you condense this to

    unit = metric.get('Unit')

I'm not so sure the comment is necessary too, can you limit that to one line?

File: leadbutt.py [hunk 2]
```diff
@@ -133,18 +135,34 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
...
+        # make sure that if we get one metric name, we make it a list and loop over
+        # the one value
+        metric_names = metric['MetricName']
```
Comment (crccheck) [inline leadbutt.py]: likewise, I think you can get rid of this comment, and if you re-write the casting code like:
```
metric_names = metric['MetricName']
if not isinstance(metric_names, list):
    metric_names = [metric_names]
```
it's more obvious what's happening

**Comment by paul-krohn** [comment]
This has been sitting for a while now, any chance of getting it merged in? I need to add a retry/backoff wrapper to avoid (undocumented) CloudWatch API request limits.

Thanks!

**Comment by crccheck** [comment]
oh wow, this has changed a lot since I last saw it. It's a bigger PR than I'd like, but that's probably my fault for not noticing it. Sorry. I think it's good enough to merge as is. I think I didn't do it earlier because the TravisCI check was failing at the time.

**Pull request merged**